addons/sale_product_configurator/controllers/main.py

```diff
@@ -47,10 +47,10 @@ def get_product_configurator_values(
             combination = request.env['product.template.attribute.value'].browse(ptav_ids).filtered(
                 lambda ptav: ptav.product_tmpl_id.id == product_template_id
             )
-            # Set attributes that are in mode `no_variant` and have only one value
+            # Set attributes that are in mode `no_variant`
             unconfigured_ptals = product_template.attribute_line_ids - combination.attribute_line_id
-            combination += unconfigured_ptals.filtered(lambda ptal: ptal.value_count == 1).mapped(
-                lambda ptal: ptal.product_template_value_ids[:1]
+            combination += unconfigured_ptals.mapped(
+                lambda ptal: ptal.product_template_value_ids._only_active()[:1]
```

Comment:
@vchu-odoo Do you remember why we needed the `filtered` ?

Reply:
The problem that we fixed that if the attribute had only one value it was not included in the combination, hence was not included in exclusions. Turned out that it was not only the case for one value, @Feyensv suggested to remove filter.